src/program_abi.rs

```diff
@@ -0,0 +1,88 @@
+#![allow(dead_code)]
```

Comment:
Putting the ignore directive here has too broad of an effect.

Opt for putting it *inside* the functions in question.

Or, write a unit test:
```rust
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn detects_enum_and_struct_types() {
        let enum_decl = type_decl_w_type_field("enum Something");
        assert!(enum_decl.is_enum_type());
        assert!(!enum_decl.is_struct_type());

        let struct_decl = type_decl_w_type_field("struct Something");
        assert!(struct_decl.is_struct_type());
        assert!(!struct_decl.is_enum_type());
    }

    fn type_decl_w_type_field(type_field: &str) -> TypeDeclaration {
        TypeDeclaration {
            type_id: 0,
            type_field: type_field.to_string(),
            components: None,
            type_parameters: None,
        }
    }
}
```

Reply:
You're right. I think I will write unit test!